scalding-core/src/main/scala/com/twitter/scalding/typed/CoGrouped.scala

```diff
@@ -333,6 +333,8 @@ abstract class CoGroupedJoiner[K](inputSize: Int,
 
   override def getIterator(jc: JoinerClosure) = {
     val iters = (0 until distinctSize).map { jc.getIterator(_).asScala.buffered }
+    // Don't pattern match in performance-critical code
```

Comment:
Whoops.  When I wrote that I assumed that Scala had some sort of safe way to get the head that returned an `Option` and that you weren't using it for performance reasons, but now I see that there is no such function in the API.  I'll change the wording to your version.